src/middlewared/middlewared/plugins/disk_/sync.py

```diff
@@ -208,18 +216,27 @@ async def sync_all(self, job):
                     # Do not issue unnecessary updates, they are slow on HA systems and cause severe boot delays
                     # when lots of drives are present
                     if self._disk_changed(disk, original_disk):
-                        await self.middleware.call('datastore.update', 'storage.disk', disk['disk_identifier'], disk)
-                        changed = True
+                        await self.middleware.call(
+                            'datastore.update', 'storage.disk', disk['disk_identifier'], disk, {'send_event': False}
+                        )
+                        changed.add(disk['disk_identifier'])
                 else:
-                    await self.middleware.call('datastore.insert', 'storage.disk', disk)
-                    changed = True
+                    await self.middleware.call('datastore.insert', 'storage.disk', disk, {'send_event': False})
+                    changed.add(disk['disk_identifier'])
 
         # make sure the database entries for enclosure slot information for each disk
         # matches with what is reported by the OS
         await self.middleware.call('enclosure.sync_disks')
 
-        if changed:
+        if changed or deleted:
             await self.middleware.call('disk.restart_services_after_sync')
+            disks = await self.middleware.call('disk.query', [], {'prefix': 'disk_'})
+            for change, delete in zip_longest(changed, deleted, fillvalue=None):
```

Comment:
What does `zip_longest` do here?

Reply:
Allows us to iterate over `change` and `delete` using  a single for loop and it also provides the `fillvalue` keyword so if the 2 variables are uneven then it won't break the loop early (like the standard `zip` does).